src/config/watcher.rs

```diff
@@ -190,19 +190,9 @@ mod tests {
     use std::{fs::File, io::Write, time::Duration};
     use tokio::sync::broadcast;
 
-    async fn test(file: &mut File, timeout: Duration, mut receiver: SignalRx) -> bool {
-        file.write_all(&[0]).unwrap();
-        file.sync_all().unwrap();
-
-        matches!(
-            tokio::time::timeout(timeout, receiver.recv()).await,
-            Ok(Ok(crate::signal::SignalTo::ReloadFromDisk))
-        )
-    }
-
-    async fn test_component_reload(
+    async fn test_signal(
         file: &mut File,
-        expected_component: &ComponentKey,
+        _expected_signal: crate::signal::SignalTo,
```

Comment:
Nit:
```suggestion
        expected_signal: crate::signal::SignalTo,
```
We reserve `_` prefix for intentionally unused variables. Does clippy complain about this?